### PR TITLE
Remove usage of global options from iptables cell

### PR DESCRIPTION
**Pull request by pippolo84**
Remove usage of `option.Config` in iptables cell. These are just leftover after the cell has been modularized.

The option `prepend-iptables-chains` used to have an env var alias named `CILIUM_PREPEND_IPTABLES_CHAIN`.  It was there to ensure backward compatibility.  ~But the alias is exactly the same string as the usual name for the option (prefix is "CILIUM_" and option name is "PREPEND_IPTABLES_CHAIN"). In other words, there was actually no alias.
Thus it is safe to move the option in the cell.Config(), even if hive package does not support additional aliasies for the same config option.~
`CILIUM_PREPEND_IPTABLES_CHAIN` alias has been deprecated with a warning and a note in the v1.15 Upgrade Notes. Read [this comment](https://github.com/cilium/cilium/pull/29088#issuecomment-1812314672) for more info.

~Depends on: https://github.com/cilium/cilium/pull/28746~

### CODE REVIEW

**Review by thorn3r — COMMENTED** [review]
changes look fairly straight-forward, but looks like there's quite a few tests failing

**Comment by pippolo84** [comment]
> changes look fairly straight-forward, but looks like there's quite a few tests failing

Missed the copy of the cell config in `newIptablesManager`, added a commit (https://github.com/cilium/cilium/pull/29088/commits/cd097802ce01f321ed41c9a0c2119193faed585f) to fix it.

**Review by asauber — COMMENTED** [review]
Re-ran failing clustermesh test

**Review by asauber — COMMENTED** [review]
Hey @pippolo84, would you be able to link to some documentation (or code) that indicates that the env var alias would be preserved?

 Alternatively, can you provide some interactive output when running the agent which shows that it is used with the new implementation?

**Comment by pippolo84** [comment]
> Hey @pippolo84, would you be able to link to some documentation (or code) that indicates that the env var alias would be preserved?
> 
> Alternatively, can you provide some interactive output when running the agent which shows that it is used with the new implementation?

Turned out that I was wrong. The alias is there for a reason.
The old code binds the env var `CILIUM_PREPEND_IPTABLES_CHAIN` (note the missing trailing `S` in the word `CHAIN`) using:

`option.BindEnvWithLegacyEnvFallback(vp, option.PrependIptablesChainsName, "CILIUM_PREPEND_IPTABLES_CHAIN")`

While, moving the option to be part of the iptables cell config, the hive package binds the env var here:

https://github.com/cilium/cilium/blob/32fbb7f186b104d52d8a7b68c06618f2f0815203/pkg/hive/hive.go#L136

and it builds the env var name to bind to from the option name, so `prepend-iptables-chains` (with the trailing `S`).

Anyway, I've discussed it offline with @aanm and setting this option through the fallback env var can be deprecated. I'll add a warning at the start of the cell and a line in the update notes.

File: Documentation/operations/upgrade.rst [hunk 1]
```diff
@@ -339,6 +339,11 @@ Annotations:
   is configured in native routing mode and :ref:`DSR mode with Geneve` is enabled.
   Explicitly configure ``--tunnel-protocol=geneve`` (or the equivalent
   ``tunnelProtocol=geneve`` helm value) when DSR with Geneve is enabled.
+* ``CILIUM_PREPEND_IPTABLES_CHAIN`` environment variable has been deprecated and
+  cannot be used anymore to enable or disable Cilium iptables rules prepend instead
+  of append.
+  Please use the ``CILIUM_PREPEND_IPTABLES_CHAINS`` environment variable (note the
+  trailing ``S``) or the command line flag ``--prepend-iptables-chains`` instead.
```
Comment (learnitall) [inline Documentation/operations/upgrade.rst]: Can we change up the wording here to include a bit of information regarding why this change was made? Maybe something like:

```suggestion
* The ``CILIUM_PREPEND_IPTABLES_CHAIN`` environment variable has been renamed
  to ``CILIUM_PREPEND_IPTABLES_CHAINS`` (note the trailing ``S``) to more accurately
  match the name of the associated command line flag ``--prepend-iptables-chains``.
```

**Review by learnitall — CHANGES_REQUESTED** [review]
Small nit on the upgrade notes, but otherwise LGTM.

**Review by thorn3r — APPROVED** [review]
aside from the open conversation, LGTM!

File: Documentation/operations/upgrade.rst (hunk 1 above)
Comment (pippolo84) [reply Documentation/operations/upgrade.rst]: Fixed, thanks!

**Review by learnitall — APPROVED** [review]
Looks good for docs, thanks!

**Review by asauber — APPROVED** [review]

**Comment by asauber** [comment]
Sounds good to me, and thanks for the link

**Comment by pippolo84** [comment]
@ldelossa gentle ping :pray:

**Review by ldelossa — APPROVED** [review]
Sorry for delay, LGTM :)
